File: hw/cañarte/u3/HW29Composite&TemplateMethod/CompositePattern/src/main/java/ec/edu/espe/compositepattern/controller/Supervisor.java
package ec.edu.espe.compositepattern.controller;
import java.util.Vector;
/**
 *
 * @author Saray Cañarte
 */

abstract class Supervisor extends Employee {
    protected Vector directReports = new Vector();

    @Override
    public void stateName() {
        super.stateName(); 
        if (directReports.size() > 0) { 
            for (int i = 0; i < directReports.size(); ++i) {
                ((Employee) directReports.elementAt(i)).stateName();
            }
        }
    }

    public void add(Employee anEmployee) {
        this.directReports.addElement(anEmployee);
    }
}

File: hw/villagomez/u3/hw29TemplateMethodAndComposite/CompositePatternJS/Supervisor.js
import { Employee } from './Employee.js';

export class Supervisor extends Employee {
    constructor() {
        super();
        this.directReports = [];
    }

    stateName() {
        super.stateName();
        if (this.directReports.length > 0) {
            this.directReports.forEach((employee) => {
                employee.stateName();
            });
        }
    }

    add(anEmployee) {
        this.directReports.push(anEmployee);
    }
}
